### PR TITLE
Asemd refactor2

### PR DESCRIPTION
**Pull request by Tetracarbonylnickel**
- fixed issue #177 
- fixed modifier bug
- checker now checking every iteration
- refactored init temp and velocities. New keyword `use_momenta` substitutes `init temp` and `init velocities `.
  - If `False`: `init temp = thermostat.temperature`
  - if `True`: MD will be started with the momenta of the structure

### CODE REVIEW

File: ipsuite/calculators/ase_md.py [hunk 1]
```diff
@@ -474,54 +467,67 @@ def run(self):  # noqa: C901
...
+                            log.critical(str(checker))
+
+                    if any(stop):
+                        break
 
                 if any(stop):
-                    self.steps_before_stopping = len(metrics_dict["energy"])
+                    self.steps_before_stopping = (
+                        idx_outer * self.sampling_rate + idx_inner
+                    )
```
Comment (PythonFZ) [inline ipsuite/calculators/ase_md.py:503]: Could we also fix https://github.com/zincware/IPSuite/issues/175 here

**Review by PythonFZ — CHANGES_REQUESTED** [review]
Looks good. Would be nice to also fix https://github.com/zincware/IPSuite/issues/175

I'll wait with merging this until the DMSO project is finished, to avoid issues with the different branches.

File: ipsuite/calculators/ase_md.py (hunk 1 above)
Comment (Tetracarbonylnickel) [reply ipsuite/calculators/ase_md.py:503]: Yes that is no problem. But what ist the plan? Just remove it or getting the information differently?

Comment (PythonFZ) [reply ipsuite/calculators/ase_md.py:503]: Just remove it. We have the length of the trajectory file that we could use instead.

Comment (Tetracarbonylnickel) [reply ipsuite/calculators/ase_md.py:503]: Yes most of the time, but the when a checker stops the simulation and `pop_last=True` the number of md steps between the two last structures is unequal to the sampling rate, and we don't know the exact simulation time.

Comment (PythonFZ) [reply ipsuite/calculators/ase_md.py:503]: @M-R-Schaefer what's your point here. I guess we can also leave the metric.

**Pull request merged**